packages/build/tests/telemetry/tests.js

```diff
@@ -127,38 +135,50 @@ test('Telemetry is disabled by default', async (t) => {
 })
 
 test('Telemetry can be enabled via flag', async (t) => {
-  const { telemetryRequests } = await runWithApiMock(t, 'success', { telemetry: true })
+  const { telemetryRequests } = await runWithApiMock(t, 'success', {
+    telemetry: true,
+    featureFlags: 'buildbot_build_telemetry',
+  })
   t.is(telemetryRequests.length, 1)
 })
 
-test('Telemetry BUILD_TELEMETRY_DISABLED env var overrides flag usage', async (t) => {
+test('Telemetry cannnot be enabled via flag if the feature flag is disabled', async (t) => {
   const { telemetryRequests } = await runWithApiMock(t, 'success', {
     telemetry: true,
-    env: { BUILD_TELEMETRY_DISABLED: 'true' },
+    flags: {},
   })
   t.is(telemetryRequests.length, 0)
 })
 
-test('Telemetry error generates no logs', async (t) => {
-  await runWithApiMock(t, 'success', { origin: 'https://...', telemetry: true, snapshot: true })
+test('Telemetry BUILD_TELEMETRY_DISABLED env var overrides flag and feature flag usage', async (t) => {
+  const { telemetryRequests } = await runWithApiMock(t, 'success', {
+    telemetry: true,
+    featureFlags: 'buildbot_build_telemetry',
+    env: { BUILD_TELEMETRY_DISABLED: 'true' },
+  })
+  t.is(telemetryRequests.length, 0)
 })
 
-test("Telemetry's unref'd child process does not disturb regular build process execution on success", async (t) => {
-  const { exitCode } = await runWithApiMock(t, 'success', {
+test('Telemetry error generates no logs', async (t) => {
+  await runWithApiMock(t, 'success', {
+    origin: 'https://...',
     telemetry: true,
-    telemetryUnrefChildProcess: true,
-    useBinary: true,
+    featureFlags: 'buildbot_build_telemetry',
     snapshot: true,
   })
-  t.is(exitCode, 0)
 })
 
-test("Telemetry's unref'd child process does not disturb regular build process execution on error", async (t) => {
-  const { exitCode } = await runWithApiMock(t, 'invalid', {
-    telemetry: true,
-    telemetryUnrefChildProcess: true,
-    useBinary: true,
-    snapshot: true,
+test('Telemetry calls timeout in less than 1.3 seconds by default', async (t) => {
+  // Start the mock telemetry server
+  const { scheme: schemeTelemetry, host: hostTelemetry, requests, stopServer } = await startServer({
+    path: TELEMETRY_PATH,
+    // eslint-disable-next-line no-magic-numbers
+    wait: 1500,
```

Comment:
This isn't ideal since we're making this a time based execution test 😕 however my idea once #2439 is merged and allows us to address #2406 is to use the thrown error produced to validate we're actually timing out.